src/random_bq_to_anndata.py

```diff
@@ -112,7 +112,7 @@ def random_bq_to_anndata(project, dataset, num_cells, output_file_prefix):
     adata.raw = adata
     adata.write(f'{output_file_prefix}.h5ad', compression="gzip")
 
-def generate_sparse_matrix(cell_data, minimum_feature_index):
+def generate_sparse_matrix(cell_data, minimum_feature_index, maximum_feature_index):
```

Comment:
I think you might be able to do this in COO format more easily… it's exactly what we're storing in BQ as opposed to a CSR format (which is complex as you're seeing!)

Reply:
Yes, thanks - I've done that. Much simpler. A shame I didn't do that *before* getting it all working using CSR.